src/main/java/seedu/address/network/Network.java

```diff
@@ -1,15 +1,43 @@
 package seedu.address.network;
 
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import org.apache.http.client.fluent.Request;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.util.StringUtil;
+
 /**
  * Manager of Network component
  */
 public abstract class Network {
+    private static final Logger logger = LogsCenter.getLogger(Network.class);
 
-    public static String fetch(String targetUrl) {
-        return lookupUrl(targetUrl);
+    public static Optional<InputStream> fetchAsStream(String url) {
+        try {
+            return Optional.of(Request.Get(url)
+                    .execute()
+                    .returnContent()
+                    .asStream());
+        } catch (IOException e) {
+            logger.warning(StringUtil.getDetails(e));
+            return Optional.empty();
+        }
     }
 
-    private static String lookupUrl(String targetUrl) {
-        return "Hello World";
+    public static String fetch(String url) {
```

Comment:
To be more explicit and consistent, I think we should rename this method as `fetchAsString`.